pkg/bloomcompactor/ringmanager.go

```diff
@@ -0,0 +1,206 @@
+package bloomcompactor
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/kv"
+	"github.com/grafana/dskit/ring"
+	"github.com/grafana/dskit/services"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// ringAutoForgetUnhealthyPeriods is how many consecutive timeout periods an unhealthy instance
+	// in the ring will be automatically removed.
+	ringAutoForgetUnhealthyPeriods = 10
+
+	// ringNameForServer is the name of the ring used by the bloom-compactor server.
+	ringNameForServer = "bloom-compactor"
+	// start with a single instance
+	ringNumTokens   = 1
+	ringCheckPeriod = 3 * time.Second
+
+	// ringKey is the key under which we register different instances of bloom-compactor in the KVStore.
+	ringKey = "bloom-compactor"
+)
+
+type RingManager struct {
```

Comment:
can we reuse one of the other implementations of ringmanager instead of recreating it here? If the logic is the same, it feels better+more extensible to reuse.